### ISSUE TITLE
[Bug]: object NoneType can't be used in 'await' expression"

### ISSUE THREAD

**Issue by barto64** [wlcrs/huawei_solar#414]
### Describe the issue

After quite a few months working more or less stable i started to have frequent disconnections forcing me to restart home assistant. As if did not solve the issue i also switch off and on the Battery, the two inverters but it did not solve the problem either. Finally I decided uninstall and install again the integration in HA. Prboel though is now I am not able to install it again, neither in my HA of PPROD nor in my HA of DEV. In the screen where the Ip is entered i get "Unexpected error" and in the HA log i can see
"[custom_components.huawei_solar.config_flow] object NoneType can't be used in 'await' expression"

### Bescribe your Huawei Solar Setup

Inverter Type: SUN2000-5KTL-TA219 and SUN2000-3.68K-HV213
Inverter Firmware version: V200R001C00SPC130
SDongle present: yes (firmware V100R001C00SPC133)
Power meter present:  single phase
Battery: LUNA2000 10kWh
Battery Firmware version: V100R002C00SPC119

### How do you connect to the inverter?

Via the SDongle, wired connection

### Upload your Diagnostics File

I cannot do it as the Integration addition fails

### Upload your relevant debug logs

```text
2023-07-12 21:23:46.867 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 43006
2023-07-12 21:23:49.254 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 47000
2023-07-12 21:23:50.437 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 47089
2023-07-12 21:23:52.237 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 30000
2023-07-12 21:23:56.000 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 30071
2023-07-12 21:23:57.616 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 37200
2023-07-12 21:23:59.001 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 47000
2023-07-12 21:23:59.794 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 47089
2023-07-12 21:24:01.033 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 37100
2023-07-12 21:24:01.961 INFO (MainThread) [custom_components.huawei_solar.config_flow] Successfully connected to inverter SUN2000-3.68KTL-L1 with SN HV2130081342
2023-07-12 21:24:01.962 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 43006
2023-07-12 21:24:03.740 DEBUG (MainThread) [huawei_solar.huawei_solar] Writing to 43006: [60]
2023-07-12 21:24:04.956 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 47000
2023-07-12 21:24:06.860 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 30000
2023-07-12 21:24:07.842 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 30071
2023-07-12 21:24:09.097 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 37200
2023-07-12 21:24:09.683 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 47000
2023-07-12 21:24:11.197 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 47089
2023-07-12 21:24:12.549 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 47954
2023-07-12 21:24:14.223 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 37100
2023-07-12 21:24:16.339 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 37125
2023-07-12 21:24:17.358 INFO (MainThread) [custom_components.huawei_solar.config_flow] Successfully connected to slave inverter 16: SUN2000-5KTL-L1 with SN TA2190014966
2023-07-12 21:24:17.359 ERROR (MainThread) [custom_components.huawei_solar.config_flow] object NoneType can't be used in 'await' expression
Traceback (most recent call last):
  File "/config/custom_components/huawei_solar/config_flow.py", line 384, in async_step_setup_network
    info = await validate_network_setup(user_input)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/huawei_solar/config_flow.py", line 159, in validate_network_setup
    await bridge.stop()
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/bridge.py", line 300, in stop
    return await self.client.stop()
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 203, in stop
    await self._client.close()
TypeError: object NoneType can't be used in 'await' expression
```

### Please confirm the following:

- [X] The problem is still present in the latest release of this integration.
- [X] I did not find an existing issue describing this problem.
- [ ] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [X] I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)

**Comment by barto64** [comment]
My sincere apologies. I look again on the closed issue, install the latest beta version as per your advised and it worked. Will see now if it is stable again.

Super thanks !!!!!

**Comment by barto64** [comment]
My sincere apologies. I look again on the closed issue, install the latest beta version as per your advised and it worked. Will see now if it is stable again.

Super thanks !!!!!

**Comment by barto64** [comment]
Fixed on #409
